### PR TITLE
Fix catch-all try/except in __main__

### PR DESCRIPTION
**Pull request by youngmit**
We were catching all exceptions, including the SystemExit exception
which is nominally raised by sys.exit(). This was leading to unexpected
stack traces.

### CODE REVIEW

**Review by ntouran — APPROVED** [review]

**Review by ntouran — CHANGES_REQUESTED** [review]
Can you add this unit test as well?

```
import unittest

from armi.__main__ import main


class TestRun(unittest.TestCase):
    def test_runCommand(self):
        """Ensure run entry point with no args completes."""
        main()
```

**Comment by youngmit** [comment]
Were you able to get this to work under pytest? I'm having a heck of a time

File: armi/tests/tutorials/c5g7-settings.locationHistory.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Assem     0     1     2     3     4     5     6     7     8     9
```
Comment (ntouran) [inline armi/tests/tutorials/c5g7-settings.locationHistory.txt:1]: don't want this committed I don't think.

**Review by ntouran — CHANGES_REQUESTED** [review]
I was only able to get it to fail in pytest. I did not try the new version with a success.

**Review by ntouran — APPROVED** [review]

**Pull request merged**